### PR TITLE
[MXNET-1407] Added test to verify Large Tensor Support for pad operator

### PR DESCRIPTION
**Pull request by access2rohit**
## Description ##
new operator supported: pad

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-1407], where 1407 refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by access2rohit** [comment]
@mxnet-label-bot add [pr-awaiting-review]

**Comment by vandanavk** [comment]
@access2rohit could you check the CI and merge conflict?

**Comment by roywei** [comment]
@access2rohit could you rebase this PR? thanks

**Comment by karan6181** [comment]
@access2rohit Could you please re-trigger the CI and resolve the merge conflicts?

**Comment by piyushghai** [comment]
@access2rohit Gentle ping...

File: tests/nightly/test_large_array.py [hunk 1]
```diff
@@ -319,6 +319,39 @@ def test_flip():
...
+    assert_almost_equal(out, np_out)
+
+def test_pad():
+    shape1 = (LARGE_X, 2, 2, 2)
+    pad1 = (0, 0, 0, 0, 1, 1, 1, 1)
+    shape2 = (LARGE_X, 2, 2, 1, 1)
+    pad2 = (0, 0, 0, 0, 1, 1, 0, 0, 0, 0)
+    # note: this op doesn't support ints yet. Add tests when supported
+    dtypes = ["float16", "float32", "float64"]
+    dtypes = ["float16"]
```
Comment (ChaiBapchya) [inline tests/nightly/test_large_array.py]: which of the above 2 is correct?

**Comment by ChaiBapchya** [comment]
@access2rohit gentle ping!

**Comment by access2rohit** [comment]
@sxjscience @apeforest @anirudh2290 @zheng-da @pengzhao-intel This PR is ready for review

File: src/operator/pad.cc [hunk 2]
```diff
@@ -558,23 +558,23 @@ void single_image_reflect_grad(const Tensor<cpu, 4, DType> &grad_in,
...
+  const index_t pad_t = pad[6];
+  const index_t pad_l = pad[8];
+  index_t iStartX = std::max(index_t{0}, -pad_l);
+  index_t iStartY = std::max(index_t{0}, -pad_t);
+  index_t iStartZ = std::max(index_t{0}, -pad_f);
+  index_t oStartX = std::max(index_t{0}, pad_l);
+  index_t oStartY = std::max(index_t{0}, pad_t);
+  index_t oStartZ = std::max(index_t{0}, pad_f);
 
   int l, ip_x, ip_y, ip_z;
```
Comment (anirudh2290) [inline src/operator/pad.cc]: why are these not changed to index_t. also why is i, j, k inside pragma omp parallel not changed to index_t.

Comment (access2rohit) [reply src/operator/pad.cc]: good catch ... missed it. Will add it now

Comment (anirudh2290) [reply src/operator/pad.cc]: can you add a test, if the current test missed catching this issue.

Comment (access2rohit) [reply src/operator/pad.cc]: I haven't added tests for backward pass since initial large tensor support only works for forward passes.

**Comment by anirudh2290** [comment]
Please open an issue for the missing tests and remaining todos in Large Tensor Support.

**Pull request merged**